src/Extraload/Transformer/MapFieldsTransformer.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Extraload\Transformer;
+
+class MapFieldsTransformer implements TransformerInterface
+{
+    private $fields;
+
+    public function __construct(array $fields)
+    {
+        $this->fields = $fields;
+    }
+
+    public function transform($data)
+    {
+        $result = [];
+        foreach ($data as $key => $val) {
+            if (array_key_exists($key, $this->fields)) {
+                $result[$this->fields[$key]] = $val;
```

Comment:
I like this transformer, but we can make it better by using https://symfony.com/doc/current/components/property_access.html and make it work with multi-dimensional arrays and objects. What do you think?

Reply:
Mmm. Let me have a look. Thank you.

Reply:
Hello there,

I believe those two may be different examples since I cannot see how the `PropertyAccess` can be used to solve this issue.

I suggested a `MapFieldsTransformer` to solve the specific need explained at https://github.com/umpirsky/Extraload/issues/18 -- perhaps the name of this transformer is not too self-explanatory at this moment -- and also because it'll be good to have a bunch of transformers performing common T tasks.

Reply:
This:

```php
    if (array_key_exists($key, $this->fields)) {
        $result[$this->fields[$key]] = $val;
```

Becomes:

```php
    $result[$this->fields[$key]] = $val;
    $propertyAccessor->setValue($result, $this->fields[$key], $val);
```

Advantage? It can set arrays, objects, private fields if there are setters... It's more powerful.

Reply:
I see, thanks for the remark.

- The `PropertyAccess` component is used now
- The name of this transformer remains the same: `MapFieldsTransformer`